### PR TITLE
Bugfix/df 1350 pdf visual fixes

### PR DESCRIPTION
**Pull request by 99999989**
- format the SV number
- remove the date part from the code
- improve the placement of the overlapping boxes

### CODE REVIEW

File: infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/pdf/ProofOfWorkEffortsPdfGenerator.java [hunk 1]
```diff
@@ -353,11 +353,15 @@ private void drawBottomHeader(PDPageContentStream stream, ProofOfWorkEfforts pro
...
-		drawHeaderBox(stream, xPosColumn2, yStart, column2Width, this.messageSource.getMessage("header.svNumber"), BOLD_FONT, proofOfWorkEfforts.getSvNumber(), false);
+		drawHeaderBox(stream, xPosColumn2, yStart, column2Width, this.messageSource.getMessage("header.svNumber"), BOLD_FONT, formatSvNumber(proofOfWorkEfforts.getSvNumber()), false);
 		drawHeaderBox(stream, xPosColumn3, yStart, column3Width, this.messageSource.getMessage("header.monthAndYear"), BOLD_FONT, String.format("%s %d", this.messageSource.getMessage("month." + proofOfWorkEfforts.getControlPeriod().getValue().getMonth().toString()), proofOfWorkEfforts.getControlPeriod().getValue().getYear()), false);
 		drawHeaderBox(stream, xPosColumn4, yStart, column4Width, this.messageSource.getMessage("header.applicationsCount"), BOLD_FONT, String.valueOf(proofOfWorkEfforts.getWorkEfforts().size()), false);
 	}
 
+	private String formatSvNumber(String svNumber) {
+		return String.format("%s.%s.%s.%s", svNumber.substring(0, 3), svNumber.substring(3, 7), svNumber.substring(7, 11), svNumber.substring(11, 13));
```
Comment (emina-str) [inline infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/pdf/ProofOfWorkEffortsPdfGenerator.java]: I see that `svNumber` can't be blank so I suppose no need to check that, but maybe to check first does the length of the string matches 13, or surround with try-catch block (IndexOutOfBoundsException)

Comment (99999989) [reply infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/pdf/ProofOfWorkEffortsPdfGenerator.java]: I've added a try-catch block.

File: infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/pdf/ProofOfWorkEffortsPdfGenerator.java [hunk 2]
```diff
@@ -353,11 +353,19 @@ private void drawBottomHeader(PDPageContentStream stream, ProofOfWorkEfforts pro
...
+		drawHeaderBox(stream, xPosColumn2, yStart, column2Width, this.messageSource.getMessage("header.svNumber"), BOLD_FONT, formatSvNumber(proofOfWorkEfforts.getSvNumber()), false);
 		drawHeaderBox(stream, xPosColumn3, yStart, column3Width, this.messageSource.getMessage("header.monthAndYear"), BOLD_FONT, String.format("%s %d", this.messageSource.getMessage("month." + proofOfWorkEfforts.getControlPeriod().getValue().getMonth().toString()), proofOfWorkEfforts.getControlPeriod().getValue().getYear()), false);
 		drawHeaderBox(stream, xPosColumn4, yStart, column4Width, this.messageSource.getMessage("header.applicationsCount"), BOLD_FONT, String.valueOf(proofOfWorkEfforts.getWorkEfforts().size()), false);
 	}
 
+	private String formatSvNumber(String svNumber) {
+		try {
+			return String.format("%s.%s.%s.%s", svNumber.substring(0, 3), svNumber.substring(3, 7), svNumber.substring(7, 11), svNumber.substring(11, 13));
+		} catch (IndexOutOfBoundsException ex) {
+			return "";
```
Comment (pavax) [inline infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/pdf/ProofOfWorkEffortsPdfGenerator.java]: I guess we should throw an IllegalStateException("Could not format svNumber")

Comment (99999989) [reply infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/pdf/ProofOfWorkEffortsPdfGenerator.java]: Done

File: infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/pdf/ProofOfWorkEffortsPdfGenerator.java [hunk 3]
```diff
@@ -353,11 +353,19 @@ private void drawBottomHeader(PDPageContentStream stream, ProofOfWorkEfforts pro
...
+		drawHeaderBox(stream, xPosColumn2, yStart, column2Width, this.messageSource.getMessage("header.svNumber"), BOLD_FONT, formatSvNumber(proofOfWorkEfforts.getSvNumber()), false);
 		drawHeaderBox(stream, xPosColumn3, yStart, column3Width, this.messageSource.getMessage("header.monthAndYear"), BOLD_FONT, String.format("%s %d", this.messageSource.getMessage("month." + proofOfWorkEfforts.getControlPeriod().getValue().getMonth().toString()), proofOfWorkEfforts.getControlPeriod().getValue().getYear()), false);
 		drawHeaderBox(stream, xPosColumn4, yStart, column4Width, this.messageSource.getMessage("header.applicationsCount"), BOLD_FONT, String.valueOf(proofOfWorkEfforts.getWorkEfforts().size()), false);
 	}
 
+	private String formatSvNumber(String svNumber) {
+		try {
+			return String.format("%s.%s.%s.%s", svNumber.substring(0, 3), svNumber.substring(3, 7), svNumber.substring(7, 11), svNumber.substring(11, 13));
+		} catch (IndexOutOfBoundsException ex) {
+			throw new IllegalStateException("Could not format svNumber");
```
Comment (pavax) [inline infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/pdf/ProofOfWorkEffortsPdfGenerator.java]: you need to pass the svNumber using string concatination in the exception message

Comment (99999989) [reply infrastructure/src/main/java/ch/admin/seco/alv/service/onlineform/infrastructure/pdf/ProofOfWorkEffortsPdfGenerator.java]: Ok, makes sense :)

**Pull request merged**